File: player/ui.py
from enum import Enum

import shared
from shared import get_image_path

IMAGES_DIR = shared.get_images_dir(__file__)


class ImagesMapping(str, Enum):
    player_paused = get_image_path(IMAGES_DIR, "Player_paused.bmp")
    player_playing = get_image_path(IMAGES_DIR, "Player_playing.bmp")
    player_playing_back = get_image_path(IMAGES_DIR, "Player_playing_back.bmp")
    player_playing_next = get_image_path(IMAGES_DIR, "Player_playing_next.bmp")
    error = get_image_path(IMAGES_DIR, "Error_player.bmp")

File: v2/player/ui.go
package player

import (
	"embed"
	"github.com/nathan-coates/desk_controller/v2/shared"
	"log"
)

//go:embed *.bmp
var images embed.FS

var (
	PlayerPaused      string
	PlayerPlaying     string
	PlayerPlayingBack string
	PlayerPlayingNext string
	PlayerError       string
)

func init() {
	mapping, err := shared.WriteImagesToFile(images)
	if err != nil {
		log.Fatal(err)
	}

	PlayerPaused = mapping["Player_paused.bmp"]
	PlayerPlaying = mapping["Player_playing.bmp"]
	PlayerPlayingBack = mapping["Player_playing_back.bmp"]
	PlayerPlayingNext = mapping["Player_playing_next.bmp"]
	PlayerError = mapping["Error_player.bmp"]
}
